backend/project/routes_webauthn_duo.py

```diff
@@ -0,0 +1,282 @@
+            webauthn_credential.public_key = str(
+                webauthn_credential.public_key, "utf-8")
+        '''
+        user = User(
+            ukey=ukey,
+            username=username,
+            display_name=display_name,
+            pub_key=webauthn_credential.public_key,
+            credential_id=webauthn_credential.credential_id,
+            sign_count=webauthn_credential.sign_count,
+            rp_id=RP_ID,
+            icon_url='https://example.com')
+        db.session.add(user)
+        db.session.commit()
+        '''
+        # Add new user entry to database table
+        dbcursor.execute('INSERT INTO users (email, display_name, ukey, icon_url, credential_id, pub_key, sign_count, rp_id) VALUES (%s, %s, %s, %s, %s, %s, %d, %s)', (email, display_name, ukey, 'https://example.com', webauthn_credential.credential_id, webauthn_credential.public_key, webauthn_credential.sign_count, RP_ID))
+        dbconnection.commit()
+    else:
+        return make_response(jsonify({'fail': 'User already exists.'}), 401)
+
+    #flash('Successfully registered with email {}.'.format(email))
+
+    return jsonify({'success': 'User successfully registered.'})
+
+
+@app.route('/verify_assertion', methods=['POST'])
+def verify_assertion():
+    challenge = session.get('challenge')
+    assertion_response = request.form
+    credential_id = assertion_response.get('id')
+
+    dbcursor.execute('SELECT email, display_name, ukey, icon_url, credential_id, pub_key, sign_count, rp_id FROM test WHERE credential_id = %s', (credential_id,))
+    user = dbcursor.fetchone()
+    if not user:
+        return make_response(jsonify({'fail': 'User does not exist.'}), 401)
+
+    webauthn_user = webauthn.WebAuthnUser(
+        user[2], # user.ukey
+        user[0], # user.username or email in our case
+        user[1], # user.display_name
+        user[3], # user.icon_url
+        user[4], # user.credential_id
+        user[5], # user.pub_key
+        user[6], # user.sign_count
+        user[7]) # user.rp_id
+
+    webauthn_assertion_response = webauthn.WebAuthnAssertionResponse(
+        webauthn_user,
+        assertion_response,
+        challenge,
+        ORIGIN,
+        uv_required=False)  # User Verification
+
+    try:
+        sign_count = webauthn_assertion_response.verify()
+    except Exception as e:
+        return jsonify({'fail': 'Assertion failed. Error: {}'.format(e)})
+
+    # Update counter.
+    user.sign_count = sign_count
+    dbcursor.execute('INSERT INTO users (email, display_name, ukey, icon_url, credential_id, pub_key, sign_count, rp_id) VALUES (%s, %s, %s, %s, %s, %s, %d, %s)', (user[0], user[1], user[2], user[3], user[4], user[5], user[6], user[7]))
+    dbconnection.commit()
+
+    # TODO: find an equivalent for login_manager
+    #login_user(user)
+
+    return jsonify({
+        'success':
+        'Successfully authenticated as {}'.format(user[0])
+    })
+
+@app.rout('/login', methods=['GET', 'POST'])
```

Comment:
```suggestion
@app.route('/login', methods=['GET', 'POST'])
```

Reply:
The typo is fixed now (f86e39fb42012226bd93866c6f25c8766bc1a8f0).